README.md

```diff
@@ -41,13 +41,39 @@ volume. Customizable parameters for volume creation include:
 | tier              | "standard"<br>"premium" | "standard"                             | storage performance tier |
 | network           | string                  | "default"                              | VPC name |
-| location          | string                  | zone where the plugin<br>is running in | zone |
 | reserved-ipv4-cidr| string		              | ""                                     | CIDR range to allocate Filestore IP Ranges from.<br>The CIDR must be large enough to accommodate multiple Filestore IP Ranges of /29 each |
 
 For Kubernetes clusters, these parameters are specified in the StorageClass.
 
 Note that non-default networks require extra [firewall setup](https://cloud.google.com/filestore/docs/configuring-firewall)
 
+## Current supported Features
+* Volume resizing: CSI Filestore driver supports volume expansion for all supported Filestore tiers.
+* Topology preferences: For better performance, it is recommended to run
+  workloads in the same zone where the Cloud Filestore instance is provisioned in.
+  For 'volumeBindingMode: WaitForFirstConsumer' mode in Storage class (and without any restriction on the allowedTopology),
+  the filestore instance will be created in the zone where the first pod, that uses it, is scheduled.
+  Optionally, zone preference can be explicitly provided by specifying 'allowedTopologies' with 'topology.gke.io/zone' key
+  in the storage class.
+  Following table lists the behavior of CreateVolume calls, w.r.t the knobs: Storage class binding mode,
+  CSI provisioner side car 'strict-topology' flag, SC allowedTopology field. The 'CSI provisioner Behavior' column describes
+  how the CreateVolume.accessibility_requirements.requisite and CreateVolume.accessibility_requirements.preferred is populated
+  by the CSI provisioner.
+  For creating the instance, Filestore driver picks the first topology in the preferred list, or the
+  first topology of the requisite list (if preferred list is empty). If topology feature is not enabled in
+  CSI provisioner (--feature-gates=Topology=false),  CreateVolume.accessibility_requirements will be nil, and the driver
+  simply creates the instance in the zone where its running.
+
+
+| SC Bind Mode         | 'strict-topology' | SC allowedTopology  | CSI provisioner Behavior    |
+| -------------------- | ----------------- | ------------------- | --------------------------- |
+| WaitForFirstCustomer |       true        |        Present      | CSI provisioner fails the provisioning if the topology of the node selected by the scheduler does not fall under SC allowedTopology list (scheduler tries to select a different node). Else, call CreateVolume with requisite = the topology of the selected node, Preferred = Requisite |
+| WaitForFirstCustomer |       false       |        Present      | Call CreateVolume with requisite = SC allowedTopology, preferred = sort and shift requiste list with selected node topology as the first parameter |
```

Comment:
Matching the above, "Call CreateVolume with requisite set to `allowedTopology` and preferred set to `allowedTopology` rearranged with the selected node topology as the first parameter. The node topology will be added if it is not in `allowedTopology`.

Reply:
from [code](https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/topology.go#L180) I see that "The node topology will NOT be added if it is not in allowedTopology".  As seen in code, selectedNodeTopology is not used, for waitForFirstCustomer, strict-topology  = false and allowedTopologies present.  requisite list is populated directly from allowedTopogies.

only the following statement is true:
"Call CreateVolume with requisite set to allowedTopology and preferred set to allowedTopology rearranged with the selected node topology as the first parameter"

Reply:
Ok, thanks for confirming. I think this supports the decision to go into some detail on how things work, clearly it's complicated.